dotnet-statsig/src/Statsig/Network/EventLogger.cs

```diff
@@ -1,52 +1,68 @@
-﻿using System.Timers;
+﻿using System;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Statsig.Network
 {
     public class EventLogger
     {
-        int _maxQueueLength;
-        SDKDetails _sdkDetails;
-        Timer _flushTimer;
-        List<EventLog> _eventLogQueue;
-        RequestDispatcher _dispatcher;
-        HashSet<string> _errorsLogged;
+        private readonly int _maxQueueLength;
+        private readonly SDKDetails _sdkDetails;
+        private readonly RequestDispatcher _dispatcher;
+        private readonly Dictionary<string, string> _statsigMetadata;
+
+        private readonly Task _backgroundPeriodicFlushTask;
+        private readonly CancellationTokenSource _shutdownCTS;
+        private DateTime _lastFlushTime = DateTime.UtcNow;
+
+        private List<EventLog> _eventLogQueue;
+        private readonly HashSet<string> _errorsLogged;
+        private readonly object _queueLock;
 
         public EventLogger(RequestDispatcher dispatcher, SDKDetails sdkDetails, int maxQueueLength = 100, int maxThresholdSecs = 60)
         {
             _sdkDetails = sdkDetails;
             _maxQueueLength = maxQueueLength;
             _dispatcher = dispatcher;
+            _statsigMetadata = new Dictionary<string, string>
+            {
+                ["sdkType"] = _sdkDetails.SDKType,
+                ["sdkVersion"] = _sdkDetails.SDKVersion,
+            };
 
             _eventLogQueue = new List<EventLog>();
             _errorsLogged = new HashSet<string>();
```

Comment:
this was planned but not currently used, and we might end up doing it another way, so I wonder if removing this `HashSet` can simplify anything?

Reply:
What do you mean "not currently used"? Are you saying that there aren't actually error events, and thus the `if (entry.IsErrorLog)` check down in `Enueue(entry)` will always be false?

As it stands, this HashSet isn't really adding any significant complexity, so I don't think we'd simplify things much by removing it.  Even without it, we would still need locking to prevent a race condition that could lead to lost events.